networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRootValidator.java

```diff
@@ -44,5 +47,7 @@ public void validate(final BlobSidecarOld blobSidecar) {
     }
 
     verifyKzg(blobSidecar);
+
+    expectedBlobIdentifiers.remove(blobIdentifier);
```

Comment:
We can just change above if to if `(!expectedBlobIdentifiers.remove(blobIdentifier))` to avoid this operation.